### PR TITLE
Make tooltip device independent interactable

### PR DESCRIPTION
**Pull request by EvitaAFD**
## [GC-1078](https://ethoslife.atlassian.net/browse/GC-1078?atlOrigin=eyJpIjoiZWRjOGY1ZWRiNmI0NDg4ODg4ZWU3MjYxMmFmZTlmYjciLCJwIjoiaiJ9)
[GC-1076](https://ethoslife.atlassian.net/browse/GC-1076?atlOrigin=eyJpIjoiZGZmMzczNmRhZmNlNDkxM2I2ZmNjZjRiNWZhNjBlYjMiLCJwIjoiaiJ9)

### Please review these reminders and either check the box or explain why not:

- [x] Read and followed the [contributing guidelines](https://eds.eks.dev.ethos-int.com/#/Guidelines?id=section-contribute)?
- [x] Component is exported from [index.js](https://github.com/getethos/ethos-design-system/blob/master/src/components/index.js)?
- [x] Type is exported from [index.d.ts](https://github.com/getethos/ethos-design-system/blob/master/src/components/index.d.ts) so Typescript users can consume it? Added a test and ran `yarn test:types`?
- [x] Bumped the yarn version?

### Referencing PR or Branch (e.g. in CMS or monorepo):

- https://ethoslife.atlassian.net/browse/GC-1076?atlOrigin=eyJpIjoiZGZmMzczNmRhZmNlNDkxM2I2ZmNjZjRiNWZhNjBlYjMiLCJwIjoiaiJ9
- https://ethoslife.atlassian.net/browse/GC-1078?atlOrigin=eyJpIjoiZWRjOGY1ZWRiNmI0NDg4ODg4ZWU3MjYxMmFmZTlmYjciLCJwIjoiaiJ9

### Screenshots and extra notes:

Screenshot 2023-01-31 at 11 48 52 AM

[GC-1078]: https://ethoslife.atlassian.net/browse/GC-1078?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
[GC-1076]: https://ethoslife.atlassian.net/browse/GC-1076?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: src/components/Tooltip/Tooltip.js [hunk 1]
```diff
@@ -115,7 +117,14 @@ export const Tooltip = ({
                 {children}
               </div>
             ) : (
-              <div className={styles.icon} ref={ref} {...referenceProps}>
+              <div
+                role="button"
```
Comment (EvitaAFD) [inline src/components/Tooltip/Tooltip.js:121]: The role is required for this to be seen as interactive and therefore allow the user to tab through the page and open the tooltip, divs shouldn't have a tabIndex on them unless you specify that they are in fact an interactive element, in this case the div is acting like a button, so this is the appropriate role.

File: src/components/Tooltip/Tooltip.js [hunk 2]
```diff
@@ -115,7 +117,14 @@ export const Tooltip = ({
...
+                className={styles.icon}
+                ref={ref}
+                aria-labelledby="info-tooltip"
+                tabIndex={0}
```
Comment (EvitaAFD) [inline src/components/Tooltip/Tooltip.js:125]: The tabIndex paired with the onFocus and onBlur are what allow users who rely solely on the keyboard to navigate the screen to be able to interact with this element

File: src/components/Tooltip/Tooltip.js [hunk 3]
```diff
@@ -115,7 +117,14 @@ export const Tooltip = ({
                 {children}
               </div>
             ) : (
-              <div className={styles.icon} ref={ref} {...referenceProps}>
+              <div
+                role="button"
+                className={styles.icon}
+                ref={ref}
+                aria-labelledby="info-tooltip"
```
Comment (t1mwillis) [inline src/components/Tooltip/Tooltip.js:124]:
```suggestion
                ariaLabelledBy="info-tooltip"
```

**Comment by t1mwillis** [comment]
Ultimately this looks good to me. A couple questions:
* Is there anyway to "check" our work here? Can we run an audit on the deploy preview?
* A little higher in the file, there are some `HEADER_ID` & `DESC_ID` variables, does it make sense to use those for our `labeledBy` value?

**Comment by EvitaAFD** [comment]
> Ultimately this looks good to me. A couple questions:
> 
> * Is there anyway to "check" our work here? Can we run an audit on the deploy preview?
> * A little higher in the file, there are some `HEADER_ID` & `DESC_ID` variables, does it make sense to use those for our `labeledBy` value?

I tested it in the styleguide, you can see the screenshot in the PR where the focus box is on the 'i' icon- you can try to test this navigating your webpage using the tab key, you can enable a focus box in your accessibility settings to help make it more obvious that an element is selected. I am not familiar with those variables - are they specifically for accessibility?

**Comment by t1mwillis** [comment]
OK -- I'll have to do some accessibility testing reading!

Those variables are are used by the "mobile" version of the tooltip, the modal. Check out line 61 in the file. Mostly just asking if it makes sense to use them for the icon, or at least declare them similarly as constants instead of the magic string `info-tooltip`. Up to you!

**Review by t1mwillis — APPROVED** [review]

**Review by t1mwillis — CHANGES_REQUESTED** [review]
Actually, let's commit that suggestion so we don't run into linter issues!

File: src/components/Tooltip/Tooltip.js [hunk 4]
```diff
@@ -14,6 +14,7 @@ import styles from './Tooltip.module.scss'
 const BREAKPOINTS = Media.BREAKPOINTS
 const HEADER_ID = 'mobile-modal-heading'
 const DESC_ID = 'mobile-modal-description'
+const ACCESSIBILTY_ID = 'info-tooltip'
```
Comment (t1mwillis) [inline src/components/Tooltip/Tooltip.js:17]:
```suggestion
const ACCESSIBILITY_ID = 'info-tooltip'
```

Comment (t1mwillis) [reply src/components/Tooltip/Tooltip.js:17]: Then let's get this in!

**Review by t1mwillis — APPROVED** [review]

**Pull request merged**